lib/c.c

```diff
@@ -724,3 +724,68 @@ void free(void *ptr)
     __freelist_head->prev = cur;
     __freelist_head = cur;
 }
+
+void snprintf(char *buffer, size_t n, char *str, ...)
+{
+    int *var_args = &str + 4;
+    int si = 0, bi = 0, pi = 0;
+
+    while (str[si]) {
+        if (bi >= n-1) {
```

Comment:
Please keep the [coding convention](https://github.com/sysprog21/shecc/blob/master/CONTRIBUTING.md#coding-convention).